Python/syntax_test_python_strings.py

```diff
@@ -648,6 +648,20 @@
 #^ meta.string.interpolated.python string.quoted.single.block.python punctuation.definition.string.begin.python
 #         ^ meta.string.interpolated.python string.quoted.single.block.python punctuation.definition.string.end.python
 
+    f"{size:.2f}"
+#    ^ meta.string.interpolated.python - meta.interpolation
+#     ^^^^^ meta.string.interpolated.python meta.interpolation.python - meta.format-spec
+#          ^^^^ meta.string.interpolated.python meta.interpolation.python meta.format-spec.python
+#              ^ meta.string.interpolated.python meta.interpolation.python - meta.format-spec
+#               ^ meta.string.interpolated.python string.quoted.double.python - meta.interpolation
+#    ^ punctuation.definition.string.begin.python
+#     ^ punctuation.section.interpolation.begin.python
+#      ^^^^ meta.qualified-name.python meta.generic-name.python
+#          ^^^^ constant.other.format-spec.python
+#          ^ punctuation.separator.format-spec.python
+#              ^ punctuation.section.interpolation.end.python
+#               ^ punctuation.definition.string.end.python
+
  rf'string'
```

Comment:
Maybe also add a `- meta.format-spec meta.format-spec` across the whole thing to catch any future scope doubling? Possibly `- punctuation punctuation` as well, but that seems less likely to happen.

Reply:
Punctuations are not assigned via `meta_scope` and thus can barely overlap.